### PR TITLE
French translation 2022 09

### PR DESCRIPTION
**Pull request by pierre-yves-monnet**

### CODE REVIEW

**Comment by mboskamp** [comment]
Hi @pierre-yves-monnet, thanks for taking another effort to finish the french translation.
I am still interested to hear why you chose not to use the apostrophe (`'`) character but instead use `\u0027`. I don't think it is necessary. Could you explain the reason behind it?

**Comment by pierre-yves-monnet** [comment]
Hello,
I used Google Translate to propose the translation; then, I just need to review it. Google translate proposed this character.
I didn't replace it.
Do be consistent, I replace this character by '

**Comment by sghidiu** [comment]
Hi Team, do we know what patch this will make it into?  Timeline?

**Comment by mboskamp** [comment]
@pierre-yves-monnet, Thanks for updating the PR, I will take a look at it this week.

@sghidiu, Only the English translation files are part of our distributions. So the changes in this PR will not affect any release. As soon as the review is done, we can merge the PR and the files are available for everybody. In fact, you could use them already. Only the English and German translations are maintained by Camunda so for all other languages the quality may vary and some of the translation files might be outdated.

File: tasklist/it.json
Comment (mboskamp) [inline tasklist/it.json:63]:
```suggestion
      "dow": 1,
      "doy": 4
```

**Review by mboskamp — CHANGES_REQUESTED** [review]
Hi @pierre-yves-monnet,

I looked through your adjustments for the French (and Italian) translations. I have left some suggestions in the respective places where I think the translation is not correct.
Additionally, I have the following remarks:
* Please don't change the order of the entries in the JSON. It's a pain to review. It is super hard to spot the actual changes. In general, the order is not important but I needed to manually sort the lines and compare them in a separate tool which is just unnecessary in my opinion.
* Please be careful with automatic solutions for translations. Always check what you actually changed in the git diff. This is easier if the entries are not shuffled around.
* I could not spot any meaningful changes in the Italian tasklist translation. There are some changes that I don't think are correct and again the order of entries is changed. Can we exclude this file from this PR?

Please address the suggestions I made. You don't need to order the lines but consider removing the Italian tasklist file and applying the comments I left.

Best,
Miklas

**Comment by mboskamp** [comment]
Hi @pierre-yves-monnet,
thanks for addressing my concerns. Here is some more input from our side: 

- Indeed it seems you deleted the Italian translation file. Please revert this commit and make sure that you only touch files that you actually want to change. It might be easier to work locally first and push your changes to the remote branch once you are done in such cases.
- After discussing this with the team we decided that it is important to us that we maintain the order of the JSON entries. It enables us to cross-check language files for missing entries and is easier not only for us but the community to maintain. If you want to use a tool to create the translation, please make sure that it respects the order of the entries.
- The commit where you applied changes that the customer asked for ([Change requested by Desjardin](https://github.com/camunda/camunda-webapp-translations/pull/159/commits/37c8cf6983859820dc73fb6005194577c3b602f5)) also deletes some entries that still exist in other languages. Can you explain why you did that?

**Comment by pierre-yves-monnet** [comment]
_Indeed it seems you deleted the Italian translation file. Please revert this commit and make sure that you only touch files that you actually want to change. It might be easier to work locally first and push your changes to the remote branch once you are done in such cases._

==> Done.
_* After discussing this with the team, we decided that it is important to us that we maintain the order of the JSON entries. It enables us to cross-check language files for missing entries and is easier not only for us but the community to maintain. If you want to use a tool to create the translation, please make sure that it respects the order of the entries. To order the entries now, I would suggest using something like Excel/Google Sheets._

 ==> Which order do you want to maintain? The same order as in the en.json? The previous order to help the diff in the PR?
   I see that the admin dictionary got a lot of changes. I check each key marked as deleted in this dictionary. There are still here, in a different order.
   The alphabetical order is a good option. Actually, the tool uses a TreeMap to have an ordered list of keys. We got this big change in admin because now it is ordered. This must occur only once (cockpit and tasklist do not have the same issue)

I checked each file manually
* admin
	- a lot of keys marked as deleted are reordoer. I checked all keys marked as deleted in the diff: they are just moving in the alphabetical order
	569 lines in en.json and fr.json
	
* cockpit: order was correct before. Mainly change UUCode by explicit code (\u0027 by ')
	2814 lines in en.json and fr.json
	
* tasklist
	Keys ended by _REFERENCE are not accurate. These are the value in English.
	513 lines in en.json and in fr.json
	
So, I think to have an alphebetical order is definitively better. I understand your concern to be able to detect each change in the commit. 
I would suggest this action plan;
1/ a first technical PR to just reorder the dictionary
2/ A second technical PR to change the Encode character to a ASCII character
3/ then this PR. All changes will be visible then.

I'm OK to go this (long) road if we can agree on that path, and if you can validate each PR in the next day.


_The commit where you applied changes that the customer asked for ([change requested by Desjardin](https://github.com/camunda/camunda-webapp-translations/pull/159/commits/37c8cf6983859820dc73fb6005194577c3b602f5)) also deletes some entries that still exist in other languages. Can you explain why you did that?_

This is correct; some keys were deleted. The tool detects the missing sentences, and it detects the "sentences not needed".
I run the tool from an older version of Camunda-bpm (now, the dictionary is not in the same repository, and it can be unsynchronized). Thank you for detecting that (thanks to the review). I replaced the missing keys.

**Comment by mboskamp** [comment]
Hi @pierre-yves-monnet,

> Which order do you want to maintain?

Please use the order as it was before your changes. This order is important for us to see what changed between two given commits. The decision was done by the development team, so please stick to this rule. We can not merge the PR otherwise.

> I checked each file manually

This is what we want to avoid by keeping the order.

> I would suggest this action plan;
1/ a first technical PR to just reorder the dictionary
2/ A second technical PR to change the Encode character to a ASCII character
3/ then this PR. All changes will be visible then.

There is no need for multiple PRs. Please bring this PR in a state that adheres to the guidelines I shared.

**Review by mboskamp — APPROVED** [review]

**Comment by mboskamp** [comment]
Hi Pierre, as discussed we will change the order of the English and German translation files. I already created an issue for that (camunda/camunda-bpm-platform#2893).

Using the ordered files, I went over the French files again. I only spotted one additional empty line but apart from that, it looks pretty good now. Let me know when you are ready and I can trigger the merge.

**Comment by pierre-yves-monnet** [comment]
Hello,
I accepted your commit. Thank you for this one.

**Pull request merged**
